File: ios/CodeScanner/ScannerViewManager.m
//
//  ScannerViewManager.m
//  RNSCanner
//
//  Created by jeff on 2018/7/14.
//  Copyright © 2018年 Facebook. All rights reserved.
//


#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE (ScannerViewManager,RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(scanValue, NSString)
RCT_EXTERN_METHOD(doScanViaManager:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(doTouchLightViaManager:(nonnull NSNumber *)node)


RCT_EXTERN_METHOD(fetchScanCode:(nonnull NSNumber *)node resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseResolveBlock)reject)

//RCT_EXTERN_METHOD(callScanViaManager:(RCTPromiseResolveBlock)resolve   rejecter:(RCTPromiseRejectBlock)reject))


@end

File: ios/CodeScanner/ScannerViewManager.swift
//
//  ScannerViewManager.swift
//  RNSCanner
//
//  Created by jeff on 2018/7/14.
//  Copyright © 2018年 Facebook. All rights reserved.
//

import Foundation


@objc(ScannerViewManager)
class ScannerViewManager : RCTViewManager {
  

  
  override func view() -> UIView! {
    return ScannerView()
  }
  
  
  
  func doScanViaManager(_ node:NSNumber) {
    print("doScanViaManager")
    DispatchQueue.main.async {
      print("dispatch Queue For DoScan")
      let sview = self.bridge.uiManager.view(forReactTag: node) as! ScannerView
      sview.doScan()
    }
  }
  
  func doTouchLightViaManager(_ node:NSNumber) {
    print("doTouchLightViaManager")
    DispatchQueue.main.async {
      print("dispatch Queue For do touch light")
      let sview = self.bridge.uiManager.view(forReactTag: node) as! ScannerView
      sview.doTouchlight()
    }
  }
  
  func sendEvent() {
    if let eventEmitter = self.bridge.module(for: EventEmitter.self) as? EventEmitter {
      eventEmitter.sendEvent(withName: "event1", body: "test")
    }
  }
  
  func fetchScanCode( _ node:NSNumber,
                      resolver resolve: @escaping RCTPromiseResolveBlock,
                      rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    
    let sview = self.bridge.uiManager.view(forReactTag: node) as! ScannerView
    
    print("fetchScanCode ")
    DispatchQueue.main.async {
      print("dispatch Queue For Fetch")
      
      sview.doScan()
    }
    
    
  }
}
